src/norm_oracles.jl

```diff
@@ -360,3 +360,64 @@ function compute_extreme_point(lmo::EllipsoidLMO, direction; v=nothing, kwargs..
     mul!(v, I, lmo.buffer, -scaling, true)
     return v
 end
+
+
+"""
+    OrderWeightNormLMO(weights,radius)
+    
+LMO with feasible set being the atomic ordered weighted l1 norm: https://arxiv.org/pdf/1409.4271
+
+```
+C = {x ∈ R^n, Ω_w(x) ≤ R} 
+```
+The weights are assumed to be positive.
+"""
+struct OrderWeightNormLMO{R,B,D} <: LinearMinimizationOracle
+    radius::R
+    mat_B::B
+    direction_abs::D
+end
+
+function OrderWeightNormLMO(weights, radius)
+    N = length(weights)
+    B = zeros(N)
+    s = zero(eltype(weights))
+    w_sort = sort(weights,rev=true)
+    for i in 1:N
+        s += w_sort[i]
+        B[i] = 1/s
+    end
+    direction_abs = similar(weights)
+    return OrderWeightNormLMO(radius,B,direction_abs)
+end
+
+function compute_extreme_point(
+    lmo::OrderWeightNormLMO,
+    direction::M;
+    v=nothing,
+    kwargs...,
+) where {M}
+    for i in eachindex(direction)
+        lmo.direction_abs[i] = abs(direction[i])
+    end
+    perm_grad = sortperm(lmo.direction_abs,rev=true)
+    max = 0
+    ind = 1
+    N = length(lmo.mat_B)
+    for i in 1:N
+        scal = 0
+        for k in 1:i
+            scal += lmo.mat_B[i]*(lmo.direction_abs[perm_grad])[k]
+        end
+        if(scal > max)
+            max = scal
+            ind = i
+        end
+    end
+    b = zeros(N)
```

Comment:
I don't think creating the vector zero is needed here. What you can do is:
1. create `v = lmo.radius .* sign.(-1*direction)`
2. for each index idx, v[idx] *= the_appropriate_b

On another note, using `sign` can have traps here, `sign(0) == 0`, which means a zero direction will give us a zero vector, which is not an extreme point of the set (0 is in the interior). Instead, you should use `signbit.(direction)`, which gives either 0 (positive) or 1 (negative), then scale that accordingly

Reply:
@victorthouvenot this is still pending right?

Reply:
Yes, but this method implies to store the vector of permutation to unsort lmo.mat_B[ind_max]. So the overall memory cost is the same.